diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -1214,6 +1307,14 @@ int DifferentialFunctionComparator::cmpOperationsWithOperands(
             Value *OpR = R->getOperand(i);
 
             if (int Res = cmpValues(OpL, OpR)) {
+                auto *UserL = dyn_cast<User>(OpL);
```

Comment:
Actually, if the new function `getReplacementValue` is implemented as proposed above, this block of code will become unnecessary since constant expressions will be handled by the `cmpValues` call.